component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/core/dao/impl/JWTAuthenticationConfigurationDAOImpl.java

```diff
@@ -0,0 +1,119 @@
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getPriority() {
+        return 10;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public JWTClientAuthenticatorConfig getPrivateKeyJWTClientAuthenticationConfigurationByTenantDomain
+    (String tenantDomain) throws JWTClientAuthenticatorServiceServerException {
+
+        try {
+
+            Resource resource = getResource(JWT_CONFIGURATION_RESOURCE_TYPE_NAME, JWT_CONFIGURATION_RESOURCE_NAME);
+            JWTClientAuthenticatorConfig JWTClientAuthenticatorConfig;
+            if (resource == null) {
+                JWTClientAuthenticatorConfig = Util.getServerConfiguration();
+            } else {
+                JWTClientAuthenticatorConfig = Util.parseResource(resource);
+            }
+            return JWTClientAuthenticatorConfig;
+        } catch (ConfigurationManagementException e) {
+            throw handleServerException(ERROR_CODE_PK_JWT_CONFIG_RETRIEVE, e, tenantDomain);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setPrivateKeyJWTClientAuthenticationConfigurationByTenantDomain
+    (JWTClientAuthenticatorConfig jwtClientAuthenticatorConfig, String tenantDomain)
+            throws JWTClientAuthenticatorServiceServerException {
+
+        try {
+            ResourceAdd resourceAdd = Util.parseConfig(jwtClientAuthenticatorConfig);
+            getConfigurationManager().replaceResource(JWT_CONFIGURATION_RESOURCE_TYPE_NAME, resourceAdd);
+        } catch (ConfigurationManagementException e) {
+            throw handleServerException(ERROR_CODE_PK_JWT_CONFIG_RETRIEVE, e, tenantDomain);
+        }
+    }
+
+    /**
+     * Retrieve the ConfigurationManager instance from the JWTServiceDataHolder.
+     *
+     * @return ConfigurationManager The ConfigurationManager instance.
+     */
+    private ConfigurationManager getConfigurationManager() {
+
+        return JWTServiceDataHolder.getInstance().getConfigurationManager();
+    }
+
+    /**
+     * Configuration Management API returns a ConfigurationManagementException with the error code CONFIGM_00017 when
+     * resource is not found. This method wraps the original method and returns null if the resource is not found.
+     *
+     * @param resourceTypeName Resource type name.
+     * @param resourceName     Resource name.
+     * @return Retrieved resource from the configuration store. Returns {@code null} if the resource is not found.
+     * @throws ConfigurationManagementException
+     */
+    private Resource getResource(String resourceTypeName, String resourceName) throws ConfigurationManagementException {
+
+        try {
+            return getConfigurationManager().getResource(resourceTypeName, resourceName);
+        } catch (ConfigurationManagementException e) {
+            if (e.getErrorCode().equals(ERROR_CODE_RESOURCE_DOES_NOT_EXISTS.getCode())) {
```

Comment:
In order to avoid possible null pointers let's swap the code like this.
`if ((ERROR_CODE_RESOURCE_DOES_NOT_EXISTS.getCode()).equals.e.getErrorCode())`

Reply:
Fixed with https://github.com/wso2-extensions/identity-oauth-addons/pull/100/commits/d07baabaa4558de269ce04e390a0674e33191160